src/impy/kinematics.py

```diff
@@ -185,57 +100,60 @@ def __init__(
         if particle1 is None or particle2 is None:
             raise ValueError("particle1 and particle2 must be set")
 
-        self.p1 = _normalize_particle(particle1)
-        self.p2 = _normalize_particle(particle2)
+        self.p1 = process_particle(particle1)
+        self.p2 = process_particle(particle2)
 
         if isinstance(self.p1, CompositeTarget):
             raise ValueError("Only 2nd particle can be CompositeTarget")
 
         p2_is_composite = isinstance(self.p2, CompositeTarget)
 
-        m1 = mass(self.p1)
+        m1 = nucleon_mass if self.p1.is_nucleus else mass(self.p1)
```

Comment:
I didn’t test what it numerically does but given that is_nucleus returns true for protons and neutrons, it will be incorrect to pick the nucleon mass instead of the correct hadron masses. :-1: to the `is_nucleus = True` for protons in `particle`.
But ok, need some workaround.

Reply:
It is slightly incorrect, yes. A condition that does not consider protons and neutrons as nuclei is
```
p.is_nucleus and not p.is_hadron
```
`is_hadron` is true for protons and neutrons, but not for "real" nuclei with A > 1. Or we use 
```
p.A and p.A > 1
```
for "real" nuclei.

Reply:
For me `p.A and p.A > 1` seems more natural and easier to understand.

Reply:
Since this distinction between p, n and "real" nuclei with A > 1 comes up a lot, I added the function `is_real_nucleus` to impy.util.

Reply:
I didn't think it would matter, but this small change in the mass actually made one of our tests trip. :)